controller/contour/controller.go

```diff
@@ -126,3 +132,33 @@ func (r *Reconciler) ensureContourRemoved(ctx context.Context, contour *operator
 	}
 	return nil
 }
+
+// contoursExist lists Contour objects in all namespaces, returning the list
+// and true if any exist other than contour.
+func (r *Reconciler) contoursExist(ctx context.Context, contour *operatorv1alpha1.Contour) (bool, *operatorv1alpha1.ContourList, error) {
+	contours := &operatorv1alpha1.ContourList{}
+	if err := r.Client.List(ctx, contours); err != nil {
+		return false, nil, fmt.Errorf("failed to list contours: %w", err)
+	}
+	if len(contours.Items) == 0 || len(contours.Items) == 1 && contours.Items[0].Name == contour.Name {
```

Comment:
Maybe add a test for this? I had to read a couple times to see what was desired.

Reply:
@stevesloka I'm working on https://github.com/projectcontour/contour-operator/pull/29 to add integration testing.

Reply:
Integration testing is good, but unit tests are also super helpful to describe these types of logic.